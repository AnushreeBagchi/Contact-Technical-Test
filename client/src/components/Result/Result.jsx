import React from "react";

import { makeStyles } from "@material-ui/core/styles";
import {
  Card,
  CardContent,
  Typography,
  Button,
  Grid,
  List,
  ListItem,
} from "@material-ui/core";

const useStyles = makeStyles({
  card: {
    height: "100%",
  },

  errorGrid: {
    color: "#7b7777",
  },

  btn: {
    marginTop: 10,
    justify: "center",
    alignItem: "center",
  },
  list : {
    fontSize :"large"
  }
});

const Result = (props) => {
  const handleClick = () => {
    props.history.push({
      pathname: "/",
    });
  };
  const classes = useStyles();
  const response = props.location && props.location.state;
  const isError = response.errors && response.errors.length ? true : false;
  const gridText = isError
    ? `Your form had below errors. Please review.`
    : `Your form submitted successfully. We will contact you soon.`;

  return (
    <Card elevation={3}>
      <CardContent>
        <Typography
          className="cardName"
          gutterBottom
          variant="h5"
          component="h2"
        >
          <Grid container justify="center" className={classes.errorGrid}>
            <Grid item xs={12} md={6} lg={6}>
              <Typography variant="h6" className={classes.title}>
                <center>{gridText} </center>
              </Typography>
              {isError ? (
                <Grid container justify="center">
                  <List>
                    {response.errors.map((error, index) => (
                      <ListItem key={index} className = {classes.list}>
                        - {error.attribute} {error.message}.
                      </ListItem>
                    ))}
                  </List>
                </Grid>
              ) : null}
            </Grid>
          </Grid>
        </Typography>
        <Button
          className="btn"
          size="medium"
          color="primary"
          onClick={() => handleClick()}
        >
          Navigate to Home
        </Button>
      </CardContent>
    </Card>
  );
};

export default Result;
