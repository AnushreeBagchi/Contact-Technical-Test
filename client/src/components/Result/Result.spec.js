import React from 'react';
import { shallow } from 'enzyme';
import Result from './Result.jsx';

describe("Tests for CustomTextField ", ()=> {
    const data = {
        state: {
            errors : [{
                "attribute": "name",
                "message" : "is missing"
            }]
        }
    }
    it('Renders correctly', () => {
        const wrapper = shallow(<Result location={data} />);
        expect(wrapper).toMatchSnapshot();
      });
})

   
   

