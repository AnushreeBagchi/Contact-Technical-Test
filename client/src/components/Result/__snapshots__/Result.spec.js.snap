// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CustomTextField  Renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  elevation={3}
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      className="cardName"
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-errorGrid-2"
        container={true}
        justify="center"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          lg={6}
          md={6}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            variant="h6"
          >
            <center>
              Your form had below errors. Please review.
               
            </center>
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Grid))
            container={true}
            justify="center"
          >
            <WithStyles(ForwardRef(List))>
              <WithStyles(ForwardRef(ListItem))
                className="makeStyles-list-4"
                key="0"
              >
                - 
                name
                 
                is missing
                .
              </WithStyles(ForwardRef(ListItem))>
            </WithStyles(ForwardRef(List))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Button))
      className="btn"
      color="primary"
      onClick={[Function]}
      size="medium"
    >
      Navigate to Home
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
